fluture.js

```diff
@@ -617,6 +658,19 @@
     return f;
   }
 
+  //Creates a dispatcher for a binary method, but takes the object first rather than last.
+  function createInvertedBinaryDispatcher(method){
+    const f = function invertedBinaryDispatch(m, a, b){
+      if(arguments.length === 1) return unaryPartial(f, m);
+      if(arguments.length === 2) return binaryPartial(f, m, a);
+      if(m && typeof m[method] === 'function') return m[method](a, b);
```

Comment:
I'm sorry. ;) It's an attempt to strike a balance between safety and performance.

The `length` checks are the fastest way I could "curry", with the performance loss of just the checks themselves if the function is called with all arguments.

The `typeof` check is just to improve the default error message you would get in that situation: `m.hook is not a function`.